clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiAdapter.groovy

```diff
@@ -176,6 +204,18 @@ class KubernetesClientApiAdapter {
     }
   }
 
+  V1PodList getPods(String namespace, Map<String, String> labels) {
+    CoreV1Api api = new CoreV1Api();
+    String label
+    if (labels != null) {
+      Map.Entry<String, String> entry = labels.entrySet().iterator().next()
+      String key = entry.getKey()
+      String value = entry.getValue()
+      label = key + "=" + value
+    }
+    api.listNamespacedPod(namespace, null, null, label, null, API_CALL_TIMEOUT_SECONDS, null)
```

Comment:
This should probably live in the `exceptionWrapper` to ensure that we collect metrics for this.